### PR TITLE
ft: S3C-2944 object legal hold functional tests

### PR DESCRIPTION
**Pull request by ilkescality**
Adds functional tests to PUT/GET Object legal hold APIs.

### CODE REVIEW

**Comment by ilkescality** [comment]
@bert-e reset

**Comment by ilkescality** [comment]
@bert-e reset

File: tests/functional/aws-node-sdk/test/object/putObjectLegalHold.js [hunk 1]
```diff
@@ -1,164 +1,164 @@
-const assert = require('assert');
-const { errors } = require('arsenal');
-const withV4 = require('../support/withV4');
-const BucketUtility = require('../../lib/utility/bucket-util');
-const removeObjectLock = require('../../lib/utility/objectLock-util');
+// const assert = require('assert');
```
Comment (miniscruff) [inline tests/functional/aws-node-sdk/test/object/putObjectLegalHold.js]: from a change perspective a single block comment might be easier here

Comment (ilkescality) [reply tests/functional/aws-node-sdk/test/object/putObjectLegalHold.js]: oh yea that's right 😅

**Review by tmacro — APPROVED** [review]

**Review by dora-korpar — APPROVED** [review]

**Pull request merged**